src/ips/algorithm.rs

```diff
@@ -74,6 +74,23 @@ impl Ips {
 
     /// Generate peer list - main function with The Algorithm
     pub async fn generate(&mut self, state: &CrunchyState) -> Vec<Peer> {
+        // Sanity check that each node is really connected to it's peers and the peers also
```

Comment:
```suggestion
        // Sanity check that each node is really connected to its peers and the peers also
```

Reply:
True 👍 Fixed.

Reply:
I'm thinking about this point:
> Added also starting check for node's connections consistency checking:
if there are any nodes connected to themselves;
if there are any nodes connected to some peers that haven't that node in their connection list.

I'm wondering if that filtering should be done during the creation of the output CrunchyState.  I.e., that it should be assumed that CrunchyState is self-consistent and well-formed?

Reply:
seems to be orthogonal to IPS

Reply:
Yes, at some point it'd be good to have consistent CrunchyState (IPS would not deal with that). That can be additional step in crunchy main, before calling IPS to filter out such self-connections or at least warn about potential asymmetrical ones.